xarray/core/accessor_str.py

```diff
@@ -129,7 +166,12 @@ def f(x):
 
         return self._apply(f)
 
-    def slice(self, start=None, stop=None, step=None):
+    def slice(
+        self,
+        start: int = None,
+        stop: int = None,
+        step: int = None,
+    ) -> Any:
```

Comment:
```suggestion
    ) -> "DataArray":
```
isn't that always a DataArray? (maybe this does not hold for all functions but most I think)

Reply:
In our usage, but there is nothing stopping someone from using this with another class with a similar API.

Reply:
The return type for `apply_ufunc` is also `Any`, so I think it would make sense to use the same thing.